src/services.py

```diff
@@ -514,131 +516,297 @@ def ban_user(self, my_user_id, other_user_id):
             )
 
 
-def serialize_chat(chat_room_result, chat_result, buyer, seller, user_id):
-    (author, dealer) = (
-        (seller, buyer) if seller.get("id") == user_id else (buyer, seller)
-    )
-    return {
-        "dealer_id": dealer.get("id"),
-        "created_at": datetime.timestamp(chat_result.get("created_at")),
-        "updated_at": datetime.timestamp(chat_result.get("updated_at")),
-        "author_name": author.get("full_name"),
-        "author_id": author.get("id"),
-        "message": chat_result.get("message"),
-        "chatRoom_id": chat_room_result.get("id"),
-    }
+class OfferService:
+    def __init__(self, config):
+        self.config = config
+
+    @staticmethod
+    def _serialize_offer(offer):
+        return {
+            "id": offer.get("id"),
+            "price": offer.get("price"),
+            "number_of_shares": offer.get("number_of_shares"),
+            "offer_status": offer.get("offer_status"),
+            "created_at": datetime.timestamp(offer.get("created_at")) * 1000,
+            "type": "offer",
+        }
+
+    @staticmethod
+    def _serialize_chat_offer(chat_room_id, offer, is_deal_closed):
+        return {
+            "chat_room_id": chat_room_id,
+            "updated_at": datetime.timestamp(offer.get("created_at")) * 1000,
+            "new_chat": OfferService._serialize_offer(offer=offer),
+            "is_deal_closed": is_deal_closed,
+        }
+
+    @staticmethod
+    def _get_current_offer(session, offer):
+        session.add(offer)
+        session.flush()
+        session.refresh(offer)
+        return offer.asdict()
+
+    @staticmethod
+    def _update_chatroom_datetime(session, chat_room, offer):
+        chat_room.updated_at = offer.get("created_at")
+        session.commit()
+
+    @staticmethod
+    def _verify_user(chat_room, user_id, user_type):
+        if (user_type == "buyer" and chat_room.buyer_id != user_id) or (
+            user_type == "seller" and chat_room.seller_id != user_id
+        ):
+            raise ResourceNotOwnedException("Wrong user")
+
+    @staticmethod
+    def _update_offer_status(session, offer, chat_room, offer_status):
+        chat_room.updated_at = offer.created_at
+        chat_room.is_deal_closed = True
+        offer.offer_status = offer_status
+        session.commit()
+
+    def create_new_offer(
+        self, chat_room_id, author_id, price, number_of_shares, user_type
+    ):
+        with session_scope() as session:
+            chat_room = session.query(ChatRoom).get(chat_room_id)
+            if chat_room == None:
+                raise ResourceNotFoundException("Chat room not found")
+            if chat_room.is_deal_closed == True:
```

Comment:
```suggestion
            if chat_room.is_deal_closed:
```

Same with some others